### ISSUE TITLE
Flexicontent universal content module, generating correct item links

### ISSUE THREAD

**Issue by ldoliv** [FLEXIcontent/flexicontent-cck#484]
Hello, I'm using the latest flexicontent version 13.

The situation is as follows:

I have a universal flexicontent module that fetches articles from category "**15:destaques**". These articles have as main category "**13:Institucional**".

I created a menu item (id 779) linking to category "**15:destaques**", the generated link is **http:// ... /destaques**

Now in the module, the link generated for an item:
**http:// ... /pt/site_content/13-institucional/25-o-algarve**

If I activate the force menu item to "779" in the module options, it generates:
![image](https://cloud.githubusercontent.com/assets/6535818/13755509/7c6aed5a-ea13-11e5-8f78-323e6d123ffc.png)
**http:// ... /pt/hoteis/adriana-beach-club-hotel-resort/destaques/13-institucional/25-o-algarve**

The correct link shoud be:
**http:// ... /pt/hoteis/adriana-beach-club-hotel-resort/destaques/item/25-o-algarve**

After inspecting helper.php on line 601, category slug is "13:institucional" which makes sense because it is the primary category, but I think should be "15:destaques", because I'm forcing to a menu item. I think the category of the menu item is being ignored. **The forced menu item should have more importance, fetching the cid from the menu item.**

To generate the proper link I had to do this:

$itemid_force	= (int)$params->get('itemid_force');
if ($itemid_force == 2) {
     $app = JFactory::getApplication();
     $menu = $app->getMenu();
     $forceMenuItem = $params->get('itemid_force_value', 0);
     $menuItem = $menu->getItem($forceMenuItem);
     $catId = $menuItem->query['cid'];
     $item->link = JRoute::_(FlexicontentHelperRoute::getItemRoute($item->_row->slug, $catId));
}

Please check if this is an issue. Thanks in advance.

**Comment by ggppdk** [comment]
Undo your changes
- and try this this test, find lines:

```
// if using CATEGORY SCOPE INCLUDE ... then link though them ... otherwise via main category
$_cl = (!$behaviour_cat && $method_cat == 3) ? 'c' : 'mc';
```

and change to
```
$_cl = 'mc';
```

does the created item link change ?
'c' is the "current category" and 'mc' is the "main category" of the item

**Comment by ldoliv** [comment]
No, It's the same:
**http:// ... /pt/hoteis/adriana-beach-club-hotel-resort/destaques/13-institucional/25-o-algarve**

**Comment by ggppdk** [comment]
My mistake i meant the reverse, please test:
$_cl = 'c';

**Comment by ldoliv** [comment]
I also tested $_cl = 'c'; but is the same.

**Comment by ggppdk** [comment]
Is the item no 25  assigned to the category no 15  (destaques)  ?

**Comment by ldoliv** [comment]
Yes, The primary category is 13 (Institucional) and secondary categories are 13 (Institucional) and 15 (Destaques).

**Comment by ggppdk** [comment]
Since you are using category scope to get items from category 
destaques

the code is currenty supposed to use it in the URL, as you see with this line:
```
// if using CATEGORY SCOPE INCLUDE ... then link though them ... otherwise via main category
$_cl = (!$behaviour_cat && $method_cat == 3) ? 'c' : 'mc';
```

**Comment by ldoliv** [comment]
Ok, thanks
